docs/configuration.md

```diff
@@ -378,6 +378,18 @@ various parts of the Emacs integration.
 In particular, `lsp-haskell` provides customization options for the `haskell-language-server`-specific parts,
 such as the path to the server executable.
 
+#### Vanilla Emacs
+
+If you're using `lsp-mode` generic plugin configuration can be added using `lsp--set-configuration`:
```

Comment:
I'm not really sure who the audience for this is: I don't think anyone is likely to be using `lsp-mode` without `lsp-haskell`. So this looks like it belongs in the `lsp-haskell` readme... and I think what you _really_ want is a bug report there pointing out that the `defcustom` options exposed by `lsp-haskell` are woefully out-of-date and incomplete... which I know about and just haven't fixed yet. Regardless, this is a useful workaround and worth documenting there.

Reply:
I mention `lsp-mode` because HIE used to mention that you had an option between it and eglot and I wasn't sure if that was still relevant here.

So as far as audience, this pertains to anyone using vanilla emacs, and there are code snippets in the docs for how to configure HLS given different environments, just, not vanilla emacs which I feel like is a common env. Also, given that there's a stack of dependencies (HLS, lsp-mode, lsp-haskell, lsp-ui, elisp), it's difficult to know how to configure HLS. I don't use the lsp for anything but haskell, and rarely touch my elisp config, so, I haven't ever needed that knowledge, and therefore configuring HLS took me weeks of avoiding, and then 1-2 hours of solving. I agree that lsp-haskell could use similar updates, and probably lsp-mode too. 

Could this be a stopgap until lsp-haskell's defcustom + documentation gets updated?